src/main/java/de/tum/in/www1/artemis/service/hestia/ExerciseHintService.java

```diff
@@ -216,7 +216,7 @@ private boolean isTaskSuccessfulInSubmission(ProgrammingExerciseTask task, Submi
         var testCasesInTask = task.getTestCases();
         var feedbacks = result.getFeedbacks();
         // the feedback can either not exist (e.g. for skipped test cases), or the attribute positive may not be set
-        return feedbacks.stream().filter(feedback -> feedback != null && testCasesInTask.stream().anyMatch(testCase -> Objects.equals(testCase.getTestName(), feedback.getText())))
+        return feedbacks.stream().filter(feedback -> feedback != null && testCasesInTask.stream().anyMatch(testCase -> Objects.equals(testCase, feedback.getTestCase())))
```

Comment:
you could change the filter to two filter calls with one (Objects::nonNull) which might be prettier